compiler/rustc_transmute/src/maybe_transmutable/mod.rs

```diff
@@ -86,6 +97,10 @@ where
         // references.
         let src = src.prune(&|def| false);
 
+        if src.is_inhabited() && !dst.is_inhabited() {
+            return Answer::No(Reason::DstUninhabited);
+        }
```

Comment:
Previously we only supported enums with an explicit repr, but such enums cannot be uninhabited. Moving our `Tree` construction to use `Layout` means we can now reason about such enums, and need to be able to report the appropriate errors when they're encountered.

See `tests/ui/transmutability/uninhabited.stderr` for what the error looks like.

Reply:
It is a SemVer stability concern, but layout stability is out-of-scope for the MVP. This trait has the same layout stability caveats as other items in the `mem` module (e.g., `size_of`, `align_of`, `offset_of`, `PointeeMetadata`); i.e., it does not connote any kind of SemVer stability.

Reply:
ok